knife/spec/support/shared/integration/knife_support.rb

```diff
@@ -57,14 +57,22 @@ def knife(*args, input: nil, instance_filter: nil)
         subcommand_class = Chef::Knife.subcommand_class_from(args)
         subcommand_class.options = Chef::Application::Knife.options.merge(subcommand_class.options)
         subcommand_class.load_deps
-        instance = subcommand_class.new(args)
+        if capture_init
+          # This ensures that knife is using our UI instance from its instantiation,
+          # which allows us to capture errors that occur during initalization
+          ui = Chef::Knife::UI.new(stdout, stderr, stdin, disable_editing: true)
+          allow(Chef::Knife::UI).to receive(:new).and_return(ui)
+          instance = subcommand_class.new(args)
+        else
+          # Default behavior doesn't ever set an expectation on Knife::UI which can interfere with caller
+          # expectations when they don't know we're already messing with that.
+          instance = subcommand_class.new(args)
+          instance.ui = Chef::Knife::UI.new(stdout, stderr, stdin, instance.config.merge(disable_editing: true))
+        end
 
         # Load configs
         instance.merge_configs
```

Comment:
I need to make sure this happens before UI setup where we merge config - though locally it did not result in any failures.

Reply:
yeah this is probably fine, its messy, but the whole KnifeSupport thing is inherently pretty messy.